lib/home_page.dart

```diff
@@ -8,15 +24,288 @@ class HomePage extends StatefulWidget {
+        if (payload['new']['read'] == true) {
+          setState(() {
+            // ignore: avoid_dynamic_calls
+            _items = _items.where((Model element) => element is Message || element.id != payload['new']['id']).toList();
+          });
+        }
+      },
+    ).subscribe();
+
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    SupabaseManager.supabaseClient.removeAllChannels();
+    super.dispose();
+  }
+
+  Future<void> load() async {
+    final List<Model> items = <Model>[];
+    final int profileId = SupabaseManager.getCurrentProfile()!.id!;
+    final List<Map<String, dynamic>> messagesData = parseHelper.parseListOfMaps(
+      await SupabaseManager.supabaseClient.from('messages').select('''
+      *,
+      sender: sender_id(*)
+      )
+    ''').eq('read', false).neq('sender_id', profileId).order('created_at'),
+    );
+    items.addAll(Message.fromJsonList(messagesData));
+    final List<Map<String, dynamic>> rideEventsData = parseHelper.parseListOfMaps(
+      await SupabaseManager.supabaseClient.from('ride_events').select('''
+      *,
+      ride: ride_id(*,
+        rider: rider_id(*),
+        drive: drive_id(*,
+          driver: driver_id(*)
+          )
+      )
+      ''').eq('read', false).order('created_at'),
+    );
+    items.addAll(RideEvent.fromJsonList(rideEventsData).where((RideEvent rideEvent) => rideEvent.isForCurrentUser()));
+    items.sort((Model a, Model b) => b.createdAt!.compareTo(a.createdAt!));
+
+    setState(() {
+      _items = items;
+      _fullyLoaded = true;
+    });
+  }
+
   @override
   Widget build(BuildContext context) {
     return Scaffold(
       appBar: AppBar(
-        title: const Text('Home'),
+        title: Text(S.of(context).pageHomeTitle),
       ),
-      body: const Center(
-        child: Text('Home'),
+      body: Padding(
+        padding: const EdgeInsets.symmetric(
+          horizontal: 20,
+          vertical: 10,
+        ),
+        child: Column(
+          children: <Widget>[
+            const SizedBox(height: 30),
+            if (_fullyLoaded)
+              Expanded(
+                child: Container(
+                  width: MediaQuery.of(context).size.width,
+                  padding: const EdgeInsets.symmetric(vertical: 10, horizontal: 10),
+                  decoration: BoxDecoration(
+                    border: Border.all(color: Theme.of(context).primaryColor),
+                    borderRadius: const BorderRadius.all(Radius.circular(5.0)),
+                  ),
+                  child: _items.isNotEmpty
+                      ? ListView.separated(
+                          itemCount: _items.length,
+                          separatorBuilder: (BuildContext context, int index) {
+                            return const SizedBox(height: 12);
+                          },
+                          itemBuilder: (BuildContext context, int index) {
+                            return _buildWidget(_items[index], context);
```

Comment:
Okay I realize this might not be an easy ask, but is it possible to group these by drives/rides? I think it's not very clear right now which notifications belong to which drives/rides (and chats are the worst offenders because they can spam the whole panel so that you have no overview at all anymore. Off the top of my head, I would have this functionality modelled after _shudder_ iOS notifications, any other opinions on this?

Reply:
I like that, but I also think that should be a follow-up PR. This is complicated enough.